### ISSUE TITLE
GPAUDITLOGGING-82 Compile fails with mongoDB plugin

### ISSUE THREAD

**Issue by robertoschwald** [robertoschwald/grails-audit-logging-plugin#91]
opened by bronoman 07/Jun/15 5:18 PM

upon adding AuditLogging plugin to my app, I get the following compile error:

23:07:33 ERROR org.codehaus.groovy.grails.web.context.GrailsContextLoaderListener - Error initializing the application: Cannot get property 'datastores' on null object
java.lang.NullPointerException: Cannot get property 'datastores' on null object
at AuditLoggingGrailsPlugin$_closure1.doCall(AuditLoggingGrailsPlugin.groovy:104)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at java.lang.Thread.run(Thread.java:745)
Process finished with exit code 1

alternative reference: 
http://stackoverflow.com/questions/23470095/grails-audit-logging-plugin-for-mongodb-is-not-working 
(same behavior)

**Comment by robertoschwald** [comment]
bronoman added a comment - 07/Jun/15 5:44 PM
affects also Grails-AuditLogging version 1.0.4

**Comment by robertoschwald** [comment]
Robert Oschwald added a comment - 08/Jun/15 4:37 PM

Is your application MongoDB-Only? Which MongoDB plugin version are you using? Maybe you can derive a DataSource.groovy example from your project.

**Comment by bronoman** [comment]
Hi, 
I'm using Grails 2.4.4 with a mixed mode DB setup. 
My AuditLogging plugin is 1.0.5
My MongoDB plugin version is: 3.0.3. with mongoDB version itself at v. 2.6.10

The mongoDB part works flawlessly. However, I cannot get the auditLogging plugin to write anything to my H2 database or mongoDB (however, I'm not sure how to configure the audit-logging plugin correcty to write to mongoDB).

Here is my DataSource.groovy file:
```groovy
// new mongo config
grails {
	mongo {
		databaseName = "noSQL" // the default database name
		host = "localhost"
		port = 27017
	}
}

// legacy H2 database, decommission mid-term
dataSource {
	pooled = true
	driverClassName = "org.h2.Driver"
	username = "**************"
	password = "****************"
	configClass = HibernateFilterDomainConfiguration
}

hibernate {
	cache.use_second_level_cache = true
	cache.use_query_cache = false
	cache.region.factory_class = 'net.sf.ehcache.hibernate.EhCacheRegionFactory' // Hibernate 3
//    cache.region.factory_class = 'org.hibernate.cache.ehcache.EhCacheRegionFactory' // Hibernate4
}

// environment specific settings
environments {
	development {
		dataSource {
			// for Hibernate Filter plugin:
			configClass = HibernateFilterDomainConfiguration
			dbCreate = "create-drop" // one of 'create', 'create-drop', 'update', 'validate', ''
			url = "jdbc:h2:mem:devDb;MVCC=TRUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE"
		}
	}
	test {
		dataSource {
			driverClassName = "com.mysql.jdbc.Driver"
			dialect = "org.hibernate.dialect.MySQL5InnoDBDialect"
			// for Hibernate Filter plugin:
			configClass = HibernateFilterDomainConfiguration
			username = "*********************"
			password = "*********************"
			dbCreate = "create-drop" // use "create", "create-drop","update"
			url = "jdbc:mysql://xxxxxxxxxxxx:xxxxxxx/xxxxxxxxxx?autoReconnect=true&useUnicode=true&characterEncoding=UTF-8"
		}
	}
	production {
		dataSource {
			driverClassName = "com.mysql.jdbc.Driver"
			dialect = "org.hibernate.dialect.MySQL5InnoDBDialect"
			// for Hibernate Filter plugin:
			configClass = HibernateFilterDomainConfiguration
			dbCreate = "update"
			url = "jdbc:mysql://xxxxxxxxxx.jelastic.dogado.eu/xxxxxxxxxxxx?autoReconnect=true&useUnicode=true&characterEncoding=UTF-8"
			username = "************"
			password = "****************"
			properties {
				maxActive = 50
				maxIdle = 25
				minIdle = 5
				initialSize = 5
				minEvictableIdleTimeMillis = 60000
				timeBetweenEvictionRunsMillis = 60000
				maxWait = 10000
				validationQuery = "/* ping */"
			}
		}
	}
}
```

**Comment by mikedehaan** [comment]
I'm getting the same error: "Cannot get property 'datastores' on null object"

The line it's failing on references the EventTriggeringInterceptor bean, however, it looks like this bean is a hibernate dependency. We're using a MongoDB only application. Per the mongodb plugin documentation, we removed the hibernate plugin from the BuildConfig:

http://grails.github.io/grails-data-mapping/mongodb/manual/guide/2.%20Getting%20Started.html#2.1%20Using%20MongoDB%20Standalone

Does this plugin work without the Hibernate plugin?

**Comment by robertoschwald** [comment]
Please test 1.0.6-SNAPSHOT.

**Comment by bronoman** [comment]
with 1.0.6-SNAPSHOT I get:
```
10:31:10 ERROR org.codehaus.groovy.grails.web.context.GrailsContextLoaderListener  - Error initializing the application: No such property: config for class: org.grails.datastore.mapping.mongo.MongoDatastore
groovy.lang.MissingPropertyException: No such property: config for class: org.grails.datastore.mapping.mongo.MongoDatastore
	at AuditLoggingGrailsPlugin$_closure1_closure2.doCall(AuditLoggingGrailsPlugin.groovy:116)
	at AuditLoggingGrailsPlugin$_closure1.doCall(AuditLoggingGrailsPlugin.groovy:113)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
note that I have in the meantime switched to full mongodb datasource:
```groovy
environments {
	development {
        println("LAUNCH: Database environment [Development]")
		grails {
			mongo {
                // see: http://docs.mongodb.org/manual/reference/connection-string/
                connectionString = "mongodb://localhost/noSQL"
 			}
		}

    }
....
```
I have not yet tested 1.0.6-SNAPSHOT in a mixed environment.

**Comment by ratancs** [comment]
Not working with MongoDB as only datastore

http://stackoverflow.com/questions/31790839/grails-audit-plugin-not-working-with-mongodb

**Comment by robertoschwald** [comment]
Is fixed in latest 1.0.6-SNAPSHOT we published AUG/13/2015. Tested with Grails 2.4.4, mongodb plugin 3.0.3 as only DS in a sample application. MongoDB server 3.0.5 on OSX.

Problem was, that the mongodb datastore does not hold a config property.

See also #99
